privacyidea/api/before_after.py

```diff
@@ -82,10 +83,19 @@
 @token_blueprint.before_app_request
 def log_begin_request():
     log.debug(u"Begin handling of request {!r}".format(request.full_path))
+    g.startdate = datetime.datetime.now()
 
 
 @token_blueprint.teardown_app_request
 def teardown_request(exc):
+    try:
+        if g.audit_object.has_data:
+            g.audit_object.finalize_log()
+    except AttributeError:
```

Comment:
is there a reason for the try...except? wouldn't `if 'audit_object' in g and g.audit_object.has_data:` suffice?

Reply:
In certain cases there is an audit_object, that is ``None``.
So it would have to be

~~~~python
    if "audit_object" in g and g.audit_object and g.audit_object.has_data:
~~~~
I thought this was a bit long, so I chose the pythonic way.